### DISCUSSION TITLE
How does 'sync --info stringArray' work during an 'argocd app sync' ?

### DISCUSSION THREAD

**Discussion by jmarcos-cano** [argoproj/argo-cd#5401] [Q&A] [answered]
I'm trying to automate our deployments using CLI as recommended from the [docs](https://argoproj.github.io/argo-cd/user-guide/ci_automation/) however `--info stringArray` does not work and does not do what I expected it to do.

I'm trying to add more info to every sync that i'm doing, adding metadata, something like ` argocd  app sync myappname --info user=username@acme.com`  and expecting it to appear in app details as shown in the image, however it appears it is not doing anything.

BTW: it is indeed synchronizing as expected

![Screen Shot 2021-02-03 at 4 51 28 PM](https://user-images.githubusercontent.com/4212422/106820104-6ac27100-6640-11eb-8da7-425cacb6ac76.png)

**Comment by alexmt** [comment, marked as answer]
Hello @jmarcos-cano ,

The `--info` flag adds info items into `app.operation.info` (which is copied to `app.status.operationState.operation.info` when operation is completed). So this is operation metadata, not application metadata. Currently, it is not shown in Argo CD UI and the feature is used to build custom integration by some users. 

Currently `argocd` CLI does not have a flag to update app level info but you can use generic `argocd app patch` command:

```
argocd app patch guestbook --patch='[{"op": "replace", "path": "/spec/info", "value": [{"name": "user", "value": "username@acme.com"}]}]'
```

**Reply by jmarcos-cano** [reply]
Thank you very much! super useful to know it.
